### DISCUSSION TITLE
Support for python interpreter free-threading mode

### DISCUSSION THREAD

**Discussion by vfdev-5** [wjakob/nanobind#690] [Q&A] [answered]
I wonder whether there are plans to provide a support for free-threading mode for python >=3.13.
Similar support is provided for pybind11: https://github.com/pybind/pybind11/pull/5148

Ref: 
- https://py-free-threading.github.io/porting/

Context:
- we would like to provide free-threading support for JAX which uses nanobind: https://github.com/google/jax/issues/23073

Thanks

**Comment by wjakob** [comment, marked as answer]
Yes, I plan to add free-threading support soon.

**Reply by vfdev-5** [reply]
Thanks for the answer @wjakob ! Would you like any help with that?

**Reply by wjakob** [reply]
I already started something in #695, but the implementation is still far from complete. An extra pair of eyes would not hurt.

**Reply by hawkinsp** [reply]
It looks good so far, and indeed it's far less invasive than I would have expected.

(Test cases that exercise nanobind concurrently are obviously needed as well.)

**Reply by wjakob** [reply]
This feature has now landed in the master branch. I will wait a bit more for it to stabilize and hear from users before releasing a new version of the library.

**Reply by vfdev-5** [reply]
Thanks @wjakob !
